packages/core/src/particle/ParticleGenerator.ts

```diff
@@ -764,4 +766,150 @@ export class ParticleGenerator {
     out.push(vertexBufferBinding);
     return index;
   }
+
+  /** @internal */
+  _calculateLocalBoundingBox(): void {
+    const { min, max } = this._renderer._localBounds;
+    const { _tempVector30: directionMax, _tempVector31: directionMin } = ParticleGenerator;
+    const lifetime = this.main.startLifetime._getMinMaxValue({ min: 0, max: 0 }).max;
+
+    // StartSpeed's impact
+    if (this.emission.shape) {
+      this.emission.shape._getStartPositionRange({ min, max });
+      this.emission.shape._getDirectionRange({ min: directionMin, max: directionMax });
+    } else {
+      min.set(0, 0, 0);
+      max.set(0, 0, 0);
+      directionMin.set(0, 0, -1);
+      directionMax.set(0, 0, 0);
+    }
+    this._adjustBoundingBoxFromDirectionalVelocity(directionMax, directionMin, lifetime, this.main.startSpeed);
+
+    // GravityModifier's impact
+    const { _tempQuat0: worldInvQuat, _tempVector32: direction } = ParticleGenerator;
+    Quaternion.invert(this._renderer.entity.transform.worldRotationQuaternion, worldInvQuat);
+    // Transform gravity direction into local space
+    direction.copyFrom(this._renderer.scene.physics.gravity);
+    Vector3.transformByQuat(direction, worldInvQuat, direction);
```

Comment:
`Vector3.transformByQuat(this._renderer.scene.physics.gravity, worldInvQuat, direction);`

Reply:
done